src/cache.rs

```diff
@@ -123,15 +124,57 @@ pub extern "C" fn load_wasm(
 
 fn do_load_wasm(
     cache: &mut Cache<GoApi, GoStorage, GoQuerier>,
-    contract_checksum: Buffer,
+    checksum: Buffer,
 ) -> Result<Vec<u8>, Error> {
-    let contract_checksum: Checksum = unsafe { contract_checksum.read() }
+    let checksum: Checksum = unsafe { checksum.read() }
         .ok_or_else(|| Error::empty_arg(CACHE_ARG))?
         .try_into()?;
-    let wasm = cache.load_wasm(&contract_checksum)?;
+    let wasm = cache.load_wasm(&checksum)?;
     Ok(wasm)
 }
 
+#[no_mangle]
+pub extern "C" fn pin(cache: *mut cache_t, checksum: Buffer, err: Option<&mut Buffer>) {
+    let r = match to_cache(cache) {
+        Some(c) => catch_unwind(AssertUnwindSafe(move || do_pin(c, checksum)))
+            .unwrap_or_else(|_| Err(Error::panic())),
+        None => Err(Error::empty_arg(CACHE_ARG)),
+    };
+    let _data = handle_c_error(r, err);
+}
+
+fn do_pin(
+    cache: &mut Cache<GoApi, GoStorage, GoQuerier>,
+    checksum: Buffer,
+) -> Result<Nothing, Error> {
```

Comment:
`Nothing`? Not `()`?

Reply:
Yeah this is needed because `handle_c_error` requires result value `T: Into<Vec<u8>>`. It was `T: Default` before, but `Checksum` has no sensible `Default`. I better fix `handle_c_error` by having an implementation for binary types and one implementation for default types.